PocketKit/Sources/PocketKit/Home/Cells/Carousel/SharedWithYouCarouselCell.swift

```diff
@@ -0,0 +1,81 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import UIKit
+import Kingfisher
+import Textile
+import SharedPocketKit
+import SharedWithYou
+
+class SharedWithYouCarouselCell: UICollectionViewCell {
+    let topView: HomeCarouselView
+
+    /// The top-most stack view, that allows to add accessory views.
+    /// If no accessory view is present, it only contains `topView`
+    private lazy var topStackView: UIStackView = {
+        let stackView = UIStackView(arrangedSubviews: [topView])
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.spacing = 8
+        return stackView
+    }()
+
+    private lazy var attributionView: SWAttributionView = {
+        let attributionView = SWAttributionView()
+        attributionView.translatesAutoresizingMaskIntoConstraints = false
+        attributionView.displayContext = .summary
+        return attributionView
+    }()
+
+    override init(frame: CGRect) {
+        topView = HomeCarouselView(frame: .zero)
+        super.init(frame: frame)
+        topView.translatesAutoresizingMaskIntoConstraints = false
+        topStackView.translatesAutoresizingMaskIntoConstraints = false
+        contentView.addSubview(topStackView)
+        contentView.pinSubviewToAllEdges(topStackView)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: configuration
+extension SharedWithYouCarouselCell {
+    func configure(with configuration: HomeCarouselCellConfiguration) {
+        topView.configure(with: configuration)
+
+        if let url = configuration.sharedWithYouUrlString {
+            Task {
+                await addAttributionView(url)
```

Comment:
so odd this comes in async.

Reply:
I guess it's because the system has to retrieve that Shared With You highlight from somewhere else that's not your app, and because of that, it's not necessarily given a high priority? 🤷‍♂️